crates/snapshot/src/snapshotter.rs

```diff
@@ -91,308 +65,129 @@ impl<DB: Database> Snapshotter<DB> {
     /// Creates a new [Snapshotter].
     pub fn new(
         provider_factory: ProviderFactory<DB>,
-        snapshots_path: impl AsRef<Path>,
-        block_interval: u64,
-    ) -> RethResult<Self> {
-        let (highest_snapshots_notifier, highest_snapshots_tracker) = watch::channel(None);
-
-        let mut snapshotter = Self {
-            provider_factory,
-            snapshots_path: snapshots_path.as_ref().into(),
-            highest_snapshots: HighestSnapshots::default(),
-            highest_snapshots_notifier,
-            highest_snapshots_tracker,
-            block_interval,
-        };
-
-        snapshotter.create_directory()?;
-        snapshotter.update_highest_snapshots_tracker()?;
-
-        Ok(snapshotter)
+        snapshot_provider: Arc<SnapshotProvider>,
+    ) -> Self {
+        Self { provider_factory, snapshot_provider }
     }
 
-    /// Ensures the snapshots directory and its temporary subdirectory are properly set up.
-    ///
-    /// This function performs the following actions:
-    /// 1. If `datadir/snapshots` does not exist, it creates it.
-    /// 2. Ensures `datadir/snapshots/temp` exists and is empty.
-    ///
-    /// The `temp` subdirectory is where snapshots are initially created before being
-    /// moved to their final location within `datadir/snapshots`.
-    fn create_directory(&self) -> RethResult<()> {
-        let temporary_path = self.snapshots_path.join(TEMPORARY_SUBDIRECTORY);
+    /// Run the snapshotter
```

Comment:
what does this do?

This must make it clear that this opens a write transaction

Reply:
added a detailed comment, and also it doesn't open a read transaction